lib/matplotlib/font_manager.py

```diff
@@ -1285,7 +1285,8 @@ def is_opentype_cff_font(filename):
 def get_font(filename, hinting_factor=None):
     if hinting_factor is None:
         hinting_factor = rcParams['text.hinting_factor']
-    return _get_font(os.fspath(filename), hinting_factor)
+    return _get_font(os.fspath(filename), hinting_factor,
+                     _kerning_factor=rcParams['text.kerning_factor'])
```

Comment:
What about folding this into `rcParams['_internal.classic_mode']`?  (how many images would need to be regen'd?)

Reply:
Well, I didn't update the non-classic tests yet, so you can see from the CI results it'd be 21 failures.

Reply:
Had a quick look at the failures.  A few of them can be improved:
- regen test_acorr, test_image_interps without any text (remove_text)
- test_titletwiny can be rewritten to a non-image-comparison, just checking the title position, like test_titlesetpos which is just below it
- for test_hatching the text labels can probably removed? (can we generate legends with empty labels? not sure...)
- I think the others images would need to be regen'd (assuming we fold the switch into _internal.classic_mode) and need to keep the text to be meaningful.

Or perhaps make this a private rcParam (leading underscore, like _internal.classic_mode)?  Or perhaps I'm just overthinking this and text.kerning_factor is fine.

Reply:
Well, I added the rcParam for downstream testing's benefits; not sure if only classic mode will be helpful there.

Reply:
Sure.  I guess my preferred solution would be _internal.old_kerning or something like that, but won't insist on it.

Reply:
I opened #15054 to track these suggestions, but for tests here I just modified the `rcParam` to use the old value.